### PR TITLE
chore: update dependencies

### PR DESCRIPTION
**Pull request by ravijayaramappa**
## Details
Routine dependency updates

## Does this PR introduce breaking changes?

* ✅ `No, it does not introduce breaking changes.`

If yes, please describe the impact and migration path for existing applications.

## The PR fulfills these requirements:
* Have tests for the proposed changes been added? ❌
* Have you followed [these instructions](../CONTRIBUTING.md#-commit-message-conventions) to clearly describe the issue being fixed or feature enhanced? ✅ 

## GUS work item
W-7258582

### CODE REVIEW

File: packages/@lwc/template-compiler/package.json [hunk 1]
```diff
@@ -21,7 +21,8 @@
     },
     "devDependencies": {
         "@types/estree": "0.0.48",
-        "@types/he": "^1.1.1"
+        "@types/he": "^1.1.1",
+        "@types/source-map": "0.5.2"
```
Comment (ravijayaramappa) [inline packages/@lwc/template-compiler/package.json:25]: `source-map` is a transitive dependency and the version used is pretty old. The newer versions have typescript support but unfortunately it does not satisfy the version required. I could override and specify the latest version of source-map in our package.json

Comment (nolanlawson) [reply packages/@lwc/template-compiler/package.json:25]: Seems safer to me to import the types for the old package than try to override with the newer package.

**Review by nolanlawson — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -21,21 +21,21 @@
...
     "@commitlint/cli": "^12.1.4",
     "@rollup/plugin-node-resolve": "^13.0.0",
     "@rollup/plugin-replace": "^2.4.1",
     "@types/acorn": "^4.0.5",
     "@types/babel__core": "^7.1.14",
     "@types/jest": "^26.0.23",
-    "@types/node": "^15.12.2",
-    "@typescript-eslint/eslint-plugin": "^4.26.1",
-    "@typescript-eslint/parser": "^4.26.1",
+    "@types/node": "^15.12.4",
```
Comment (pmdartus) [inline package.json:33]: Theoretically, we are still using `node@14` but I guess that it doesn't no harm to use the `node@15` types.

File: yarn.lock [hunk 3]
```diff
@@ -16,11 +16,23 @@
   dependencies:
     "@babel/highlight" "^7.12.13"
 
+"@babel/code-frame@^7.14.5":
```
Comment (pmdartus) [inline yarn.lock:19]: Something looks wrong in the `yarn.lock` as all the babel dependencies are now duplicated.
How do you update the packages?

I would recommend running `yarn upgrade` to get rid of those duplicated packages.

File: packages/@lwc/template-compiler/package.json (hunk 1 above)
Comment (pmdartus) [reply packages/@lwc/template-compiler/package.json:25]: I don't understand why using `source-map` is needed here? Where do we use `source-map` in the template compiler?

Comment (nolanlawson) [reply packages/@lwc/template-compiler/package.json:25]: `yarn why source-map` shows multiple packages rely on it

<details><summary>click here</summary>

```
=> Found "source-map@0.6.1"
info Has been hoisted to "source-map"
info Reasons this module exists
   - "workspace-aggregator-b5b0bfa9-908a-412a-b018-95c6433d59ee" depends on it
   - Hoisted from "_project_#integration-karma#karma#source-map"
   - Hoisted from "_project_#jest#@jest#core#@jest#reporters#source-map"
   - Hoisted from "_project_#rollup-plugin-terser#terser#source-map-support#source-map"
   - Hoisted from "_project_#jest#@jest#core#@jest#transform#source-map"
   - Hoisted from "_project_#integration-karma#karma-coverage#istanbul-lib-source-maps#source-map"
   - Hoisted from "_project_#jest#@jest#core#jest-runtime#@jest#source-map#source-map"
   - Hoisted from "_project_#lerna#@lerna#version#@lerna#conventional-commits#conventional-changelog-core#conventional-changelog-writer#handlebars#source-map"
   - Hoisted from "_project_#jest#@jest#core#jest-config#jest-environment-jsdom#jsdom#escodegen#source-map"
```
</details>

I guess it's technically not the template compiler that needs it? So maybe we should just put it in the top-level package.json?

File: yarn.lock (hunk 3 above)
Comment (ravijayaramappa) [reply yarn.lock:19]: I usually run “ yarn upgrade-interactive --latest”, followed by “yarn install” to take care of any dependencies I manually update.

File: packages/@lwc/template-compiler/package.json (hunk 1 above)
Comment (ravijayaramappa) [reply packages/@lwc/template-compiler/package.json]: `@types/source-map` was not required after all. Running `yarn upgrade` seems to have fixed the source-map version at 0.6.1 which has typescript support.

**Review by pmdartus — APPROVED** [review]

**Pull request merged**